### PR TITLE
[Crash fix] - removed unnecessary microphone device type

### LINKED ISSUE #14174: SIGABRT: %{public}s > NSMicrophoneUsageDescription > This app has crashed because it attempted to access p...
Sentry Issue: [WOOCOMMERCE-IOS-1MP0](https://a8c.sentry.io/issues/5227823455/?referrer=github_integration)

```
SIGABRT: %{public}s > NSMicrophoneUsageDescription > This app has crashed because it attempted to access privacy-sensitive data without a usage description.  The app's Info.plist must contain an NSMicrophoneUsageDescription key with a string value explaining to the user how the app uses this data.
  ?, in __abort_with_payload
  ?, in abort_with_payload_wrapper_internal
  ?, in abort_with_payload
  ?, in __TCC_CRASHING_DUE_TO_PRIVACY_VIOLATION__
  ?, in ___tcc_server_send_request_authorization_block_invoke_3
...
(7 additional frame(s) were not displayed)
```

@rachelmcr was able to reproduce this:
- Go to the Hub Menu.
- Open WooCommerce Admin.
- In the admin dashboard, open the site’s Media section.
- Tap the button to add new media.
- Select “Take Photo or Video.”
- In the device camera, switch to Video.

### PR DESCRIPTION
**Pull request by bozidarsevo**
Closes: #14174

## Description

There was an unnecessary `.builtInMicrophone` device type included in the capture device list for the code scanner, which caused crashes for some users.:
`NSMicrophoneUsageDescription > This app has crashed because it attempted to access privacy-sensitive data without a usage description.  The app's Info.plist must contain an NSMicrophoneUsageDescription key with a string value explaining to the user how the app uses this data.`

Since we do not need that device type, it is removed now.

## Testing information

- Run the app on the physical device, iPhone or iPad
- Open Products tab
- Tap the scan button in top left
- Check that the swap camera button is shown in the code scanner
- Tap the swap camera button few times and make sure that the camera is actually being swapped and you are getting proper camera feed, without getting app to crash

## Screenshots

https://github.com/user-attachments/assets/f7ab02b8-9c61-48a1-9961-5eb3369b2605

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [ ] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.
- [ ] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.
- [ ] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on all devices (phone/tablet) and no regressions are added.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/Classes/ViewRelated/Products/SKU Scanner/CodeScannerViewController.swift [hunk 1]
```diff
@@ -75,7 +75,9 @@ final class CodeScannerViewController: UIViewController {
     }
 
     private func configureSwapCameraButton() {
-        swapCameraButton.isHidden = captureDevice(with: .front) == nil || captureDevice(with: .back) == nil
+        let isFrontCameraAvailable = captureDevice(with: .front) != nil
+        let isBackCameraAvailable = captureDevice(with: .back) != nil
+        swapCameraButton.isHidden = !(isFrontCameraAvailable && isBackCameraAvailable)
```
Comment (bozidarsevo) [inline WooCommerce/Classes/ViewRelated/Products/SKU Scanner/CodeScannerViewController.swift:80]: Also added this to make it more readable and understandable

**Comment by iamgabrielma** [comment]
👋 Thanks for addressing this! Were you able to replicate the crash? 

I'm not 100% sure if this change is enough to avoid it or would only mitigate it: Aside from removing `.builtInMicrophone` as an available device, I think we should also remove the force-unwrap from `CodeScannerViewController.swapCamera` when we declare `AVCaptureDeviceInput`, as we're adding unnecessary risk of force-wrapping a nil reference (specially since access to these can be modified by the user on runtime). Eg:
```
    // Ensure newDevice is not nil
    guard let validDevice = newDevice else { 
        print("Failed to find a new camera device.")
        return
    }
``` 

Then we only call `.removeInput` and `.addInput` inside of the do block, or catch the error

```
    do {
        let deviceInput = try AVCaptureDeviceInput(device: validDevice)
        session.removeInput(input)
        session.addInput(deviceInput) ... 

```

**Comment by bozidarsevo** [comment]
> Were you able to replicate the crash?

Unfortunately not (did not find reproducing steps and manually was not able to reproduce), but looks like that initial change and the last one should cover it.

**Comment by iamgabrielma** [comment]
> Unfortunately not (did not find reproducing steps and manually was not able to reproduce), but looks like that initial change and the last one should cover it.

Gotcha! 🚢

**Review by iamgabrielma — APPROVED** [review]

**Pull request merged**